src/ext/ws.js

```diff
@@ -122,12 +122,7 @@ This extension adds support for WebSockets to htmx.  See /www/extensions/ws.md f
 			return htmx.createWebSocket(wssSource)
 		});
 
-		socketWrapper.addEventListener('message', function (event) {
-			if (maybeCloseWebSocketSource(socketElt)) {
-				return;
-			}
-
-			var response = event.data;
+		var responseHandler = function (response) {
 			if (!api.triggerEvent(socketElt, "htmx:wsBeforeMessage", {
```

Comment:
The initial idea behind supporting binary formats was to give users simple way to manage websocket connections, while leaving actual message processing to their custom JS via events. By forcibly converting everything to strings, we disable this path for those who walk it. It's also a breaking change

I think, there are two ways to compromise:

1. Put this behavior behind configuration value, `false` by default
2. Offload this responsibility to event handler by allowing to modify `message` field below

Reply:
Ah, my net was too wide. The `makeFragment` function assumes it is receiving a string and that is the code path I was having an issue with. I've changed my solution so the response only forcibly converts the response to a string for that function.